features/feature_case/thrust/thrust-math.cu

```diff
@@ -9,85 +9,217 @@
+// {
+//   auto c = thrust::polar(5.48569f, 1.33698f);
+//   c = thrust::pow(c, 1.0f);
+//   *res = thrust::abs(c);  
+// }
+
+void complex_pow3(float * res)
+{
+  auto c = thrust::polar(11.48569f, 1.33698f);
   c = thrust::pow(c, c);
+  *res = thrust::abs(c);  
+}
+
+void complex_sin(float * res)
+{
+  auto c = thrust::polar(11.48569f, 1.33698f);
   c = thrust::sin(c);
+  *res = thrust::abs(c);  
+}
+
+void complex_cos(float * res)
+{
+  auto c = thrust::polar(1.48569f, 1.33698f);
   c = thrust::cos(c);
-  c = thrust::tan(c);
-  c = thrust::asin(c);
-  c = thrust::acos(c);
-  c = thrust::atan(c);
-  c = thrust::sinh(c);
-  c = thrust::cosh(c);
-  c = thrust::tanh(c);
-  c = thrust::asinh(c);
-  c = thrust::acosh(c);
-  c = thrust::atanh(c);
-  c = thrust::log(c);
-  c = thrust::exp(c);
-  c = thrust::proj(thrust::norm(c));
-  c = thrust::conj(c);
   *res = thrust::abs(c);  
 }
 
 bool test_math(){
   float *hostRes = (float *)malloc(sizeof(float));
   float *Res = (float *)malloc(sizeof(float));
   float *deviceRes;
+  bool flag = true;
   cudaMalloc((float **)&deviceRes, sizeof(float));
-  complex_math_kernel<<<1, 1>>>(deviceRes);
+
+  // complex_log10_kernel<<<1, 1>>>(deviceRes);
+  // cudaDeviceSynchronize();
+  // cudaMemcpy(hostRes, deviceRes, sizeof(float), cudaMemcpyDeviceToHost);
+  // cudaDeviceSynchronize();
+  // complex_log10(Res);
+  // if(std::abs(*hostRes-*Res)>1e-6){
+  //   std::cout<<"complex_log10 "<<" test failed \n";
+  //   std::cout<<*hostRes<<'\t'<<*Res<<'\n';
+  // }
+  // else{
+  //   std::cout<<"complex_log10 "<<"test pass \n";
+  // }
+  // flag = flag & (std::abs(*hostRes-*Res)<1e-6);
+
+  complex_sqrt_kernel<<<1, 1>>>(deviceRes);
+  cudaDeviceSynchronize();
   cudaMemcpy(hostRes, deviceRes, sizeof(float), cudaMemcpyDeviceToHost);
-  cudaFree(deviceRes);
-  complex_math(Res);
+  cudaDeviceSynchronize();
+  complex_sqrt(Res);
   if(std::abs(*hostRes-*Res)>1e-6){
+    std::cout<<"complex_sqrt "<<" test failed \n";
     std::cout<<*hostRes<<'\t'<<*Res<<'\n';
-    std::cout<<"test failed \n";
-    free(hostRes);
-    free(Res);
-    return false;
   }
   else{
-    std::cout<<"test pass \n";
-    free(hostRes);
-    free(Res);
-    return true;
+    std::cout<<"complex_sqrt "<<"test pass \n";
+  }
+  flag = flag & (std::abs(*hostRes-*Res)<1e-6);
+
+  complex_pow1_kernel<<<1, 1>>>(deviceRes);
+  cudaDeviceSynchronize();
+  cudaMemcpy(hostRes, deviceRes, sizeof(float), cudaMemcpyDeviceToHost);
+  cudaDeviceSynchronize();
+  complex_pow1(Res);
+  if(std::abs(*hostRes-*Res)>1e-6){
+    std::cout<<"complex_pow1 "<<" test failed \n";
+    std::cout<<*hostRes<<'\t'<<*Res<<'\n';
   }
+  else{
+    std::cout<<"complex_pow1 "<<"test pass \n";
+  }
+  flag = flag & (std::abs(*hostRes-*Res)<1e-6);
+
+  // complex_pow2_kernel<<<1, 1>>>(deviceRes);
+  // cudaDeviceSynchronize();
+  cudaMemcpy(hostRes, deviceRes, sizeof(float), cudaMemcpyDeviceToHost);
+  cudaDeviceSynchronize();
+  // complex_pow2(Res);
```

Comment:
Please remove these unused code.

Reply:
not unused, it will be re-enabled after intel/llvm fix related API.
Leave a mark